### PR TITLE
322 Replace local path links within documentation with sphinx 'ref' links

### LINKED ISSUE #321: Add Links in Getting Started --> Expert User Installation
# Requested Update

### Description
Currently, in the sphinx documentation, 'Getting Started' --> 'Expert User Installation' section, there are references to the 'Complete Local conda-based GeoIPS Installation <./installation.rst>', however they aren't linked to any actual documentation. I found this confusing, and it should be an easy fix.

### Background and Motivation
Improve ease of use for the GeoIPS install process. By adding these links, users will easily be able to hop back and forth between documentation for a streamlined installation process.

### Alternative Solutions
* Alternatively, we can add a subdirectory in this section which contains the aforementioned information. This would be repetitive however.

### LINKED ISSUE #322: Add Links Throughout GeoIPS Sphinx Documentation
# Requested Update

### Description
In some portions of the GeoIPS Sphinx documentation, there are references to installation documentation or other web documentation, etc. which don't have any links to these docs. These should be updated accordingly for ease of use of the people using these docs.

### Background and Motivation
By adding links, we prevent the user having to manually locate or search these sites to find the appropriate supporting documentation.

### PR DESCRIPTION
**Pull request by evrose54**
# Reviewer Checklist

* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
  * [x] None required (explain if checked, delete line if not checked)
* [x] Required ***unit tests*** added and pass for new/modified functionality
  * [x] None required (explain if checked, delete line if not checked)
* [x] Required ***integration tests*** added and pass for new/modified functionality
  * [x] None required. (explain if checked, delete line if not checked)
* [x] Required ***documentation*** added for new/modified functionality
  * [x] None required (explain if checked, delete line if not checked)
* [x] Required ***release notes*** added for new/modified functionality
  * [x] None required. (explain if checked, delete line if not checked)

Note: I've yet to add changes into release notes as I'm waiting for the next release_notes.rst file to be created.

https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes #321 
fixes #322 

# Testing Instructions
To test the new documentation updates, run this command, replacing $geoips_repopath with your path to the top level
of the geoips folder, and $geoips_pkgname with 'geoips'

$geoips_repopath/docs/build_docs.sh $geoips_repopath $geoips_pkgname html_only

# Summary
This PR had been created to add links to corresponding sections throughout GeoIPS documentation. I may have missed some links, but after checking a few times I think it's all in order. 
Added links to:
  - geoips/docs/source/starter/expert_installation.rst
  - geoips/docs/source/started/extending.rst
  - geoips/docs/source/starter/installation.rst
  - geoips/docs/source/userguide/plugin_extend.rst

# Output
To view the output, download the geoips/build/sphinx/html directory, and copy your index.html file into your browser.

### CODE REVIEW

File: docs/source/starter/installation.rst [hunk 1]
```diff
@@ -10,6 +10,8 @@
  | # # # for more details. If you did not receive the license, for more information see:
  | # # # https://github.com/U-S-NRL-Marine-Meteorology-Division/
 
+.. _installation:
```
Comment (mindyls) [inline docs/source/starter/installation.rst:13]: This is named differently in the other documentation PR

**Review by mindyls — DISMISSED** [review]
Were these links just not working before?  Note you will have to deconflict with the other PR (and did that other PR resolve some of these as well?), but otherwise I think it is fine.

File: docs/source/starter/expert_installation.rst [hunk 2]
```diff
@@ -14,7 +14,7 @@
 Expert User Installation (Administrative privileges)
 ****************************************************
 
-The `Complete Local conda-based GeoIPS Installation <./installation.rst>`
+The :ref:`Complete Local conda-based GeoIPS Installation<installation>`
```
Comment (mindyls) [inline docs/source/starter/expert_installation.rst:17]: I guess these links were originally just set up to link within the github rendered RST pages, and were broken links within the html documentation?

**Review by mindyls — APPROVED** [review]
This looks good to me.  Ready for merge with another approval.

File: docs/source/starter/installation.rst (hunk 1 above)
Comment (mindyls) [reply docs/source/starter/installation.rst]: Will deconflict from the other PR.  Resolving.

File: docs/source/starter/expert_installation.rst (hunk 2 above)
Comment (mindyls) [reply docs/source/starter/expert_installation.rst]: Yeah, they didn't even work rendered directly within github...

**Review by cpcamacho — APPROVED** [review]

**Pull request merged**